src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusCollectionManager.cs

```diff
@@ -168,11 +168,13 @@ private void WaitForReadersToComplete()
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    private bool ExecuteCollect()
+    private bool ExecuteCollect(bool openMetricsRequested)
     {
         this.exporter.OnExport = this.onCollectRef;
+        this.exporter.OpenMetricsRequested = openMetricsRequested;
         var result = this.exporter.Collect(Timeout.Infinite);
         this.exporter.OnExport = null;
+        this.exporter.OpenMetricsRequested = false;
```

Comment:
Is this needed? We always set `this.exporter.OpenMetricsRequested = openMetricsRequested;` before calling `Collect`.

Reply:
No I guess you're right, we don't need this.